processing/src/main/java/org/apache/druid/query/aggregation/post/FieldAccessPostAggregator.java

```diff
@@ -78,10 +89,28 @@ public String getName()
     return name;
   }
 
+  @Override
+  public ValueType getType()
+  {
+    return type;
+  }
+
   @Override
   public FieldAccessPostAggregator decorate(Map<String, AggregatorFactory> aggregators)
   {
-    return this;
+    final ValueType type;
+
+    if (aggregators != null && aggregators.containsKey(fieldName)) {
+      type = aggregators.get(fieldName).getType();
+    } else {
+      type = ValueTypes.defaultAggregationType();
```

Comment:
Why not keep it null here? What are the pros and cons of null vs `ValueTypes.defaultAggregationType()`?

Reply:
Ah, I think it should be `null`, this is a leftover from before i made `PostAggregator.getType` be annotated `@Nullable`.